src/main/java/org/skriptlang/skript/lang/simplification/SimplifiedLiteral.java

```diff
@@ -0,0 +1,52 @@
+package org.skriptlang.skript.lang.simplification;
+
+
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.util.ContextlessEvent;
+import ch.njol.skript.lang.util.SimpleLiteral;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a literal, i.e. a static value like a number or a string, that was created by simplifying another expression.
+ * Maintains a reference to the original expression for proper toString values.
+ * @param <T> the type of the literal
+ */
+public class SimplifiedLiteral<T> extends SimpleLiteral<T> {
+
+	public static <T> SimplifiedLiteral<T> fromExpression(Expression<T> original) {
```

Comment:
Docs? It would be good to make clear that this assumes no special context (i.e. specific events) are needed to obtain the values of 'original'